### PR TITLE
oauth: Fix TypeError

### PR DESCRIPTION
**Pull request by curioussavage**
The refresh request is not working and throwing errors
because we were not checking if the token existed before
accessing properties

:eyeglasses: @schwers or @ajacksified or @nramadas

### CODE REVIEW

File: assets/js/clientLib.es6.js [hunk 1]
```diff
@@ -120,28 +120,32 @@ export function refreshToken (app) {
...
+      if (token && token.token) {
+        const now = new Date();
+        const expires = new Date(token.tokenExpires);
 
-      Object.assign(app.getState('ctx'), {
-        token: token.token,
-        tokenExpires: token.tokenExpires,
-      });
+        Object.assign(app.getState('ctx'), {
+          token: token.token,
```
Comment (nramadas) [inline assets/js/clientLib.es6.js]: https://en.wikipedia.org/wiki/Semantic_satiation

Comment (curioussavage) [inline assets/js/clientLib.es6.js]: yap...

Comment (curioussavage) [inline assets/js/clientLib.es6.js]: ill defer improvements to that to a future pr though

**Comment by curioussavage** [comment]
looks like this might require some more work. I was assuming a retry would occur but we might need to manually retry and or refresh the page if the token already expired

**Comment by kjoconnor** [comment]
If you have any OAuth questions please let me know! Especially whatever pain points you are running into. I'm working with the mobile team on OAuth stuff so happy to just be a reference if needed.

**Comment by ajacksified** [comment]
:fish: - better than current, but we need a good solution to handle failed api requests that don't involve routing. probably snoode v2.

**Pull request closed without merging**